From bc96d7d6a32fc690f2f29d56d3ccc7dbffa90947 Mon Sep 17 00:00:00 2001
From: Sailaja <sbandarupalli@habana.ai>
Date: Thu, 05 Aug 2021 08:46:44 +0300
Subject: [PATCH] [SW-50869] gaudi2: add RST_SRC causes

for Gaudi2, PSOC_GLOBAL_CONF_RST_SRC register contain more reset causes
Added code to print gaudi2 specific reset causes

Signed-off-by: Sailaja <sbandarupalli@habana.ai>
Change-Id: I52c108ac86157dbb40e20e6de6fe1cebe387cd54
---

diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index f63e1c0..84a2cfa 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -536,3 +536,46 @@
 	regmap_writel(ASIF_SLV_ASIF_SLV_CFG_REG_VAL,
 			mmDCORE3_XBAR_DMA_PLL_ASIF_SLV_ASIF_SLV_CFG);
 }
+
+void rst_src_dump(void)
+{
+	unsigned int rst_src = regmap_readl(mmPSOC_GLOBAL_CONF_RST_SRC);
+	const char *cause;
+
+	switch (rst_src) {
+	case HL_COLD_RST:
+		cause = "cold";
+		break;
+	case HL_MANUAL_RST:
+		cause = "manual";
+		break;
+	case HL_PRSTN_RST:
+		cause = "prstn";
+		break;
+	case HL_SOFT_RST:
+		cause = "soft";
+		break;
+	case HL_WD_RST:
+		cause = "watch dog";
+		break;
+	case HL_FW_ALL_RST:
+		cause = "firmware all";
+		break;
+	case HL_SW_ALL_RST:
+		cause = "software all";
+		break;
+	case HL_FLR_RST:
+		cause = "functional";
+		break;
+	case HL_ECC_DERR_RST:
+		cause = "ecc derr";
+		break;
+	default:
+		cause = "unknown";
+		break;
+	}
+	preboot_puts_raw("Last reset cause is: ");
+	preboot_puts_raw(cause);
+	preboot_puts(" reset\n");
+}
+
diff --git a/arch/gaudi2/common/hl.h b/arch/gaudi2/common/hl.h
index a2f7240..a581a80 100644
--- a/arch/gaudi2/common/hl.h
+++ b/arch/gaudi2/common/hl.h
@@ -7,11 +7,15 @@
 #define ARCH_GAUDI2_COMMON_HL_H_
 
 enum hl_rst_src {
-	HL_COLD_RST,
-	HL_MANUAL_RST,
-	HL_PRSTN_RST = 2,
-	HL_WD_RST = 4,
-	HL_SW_ALL_RST = 8,
+	HL_COLD_RST = 1,
+	HL_MANUAL_RST = 2,
+	HL_PRSTN_RST = 4,
+	HL_SOFT_RST = 8,
+	HL_WD_RST = 16,
+	HL_FW_ALL_RST = 32,
+	HL_SW_ALL_RST = 64,
+	HL_FLR_RST = 128,
+	HL_ECC_DERR_RST = 256
 };
 
 void set_pci_msix_pba_offset_val(void);
